### PR TITLE
Add a context menu example

### PR DESCRIPTION
**Pull request by willgates**
- [x] Loading the examples from the scripts folder
- [x] Add the context menu component
- [x] Add a context menu example
- [x] Tests for new components/functionality

### CODE REVIEW

File: src/Cell.js [hunk 1]
```diff
@@ -82,6 +82,13 @@ const Cell = React.createClass({
     }
   },
 
+  onCellContextMenu() {
+    let meta = this.props.cellMetaData;
+    if (meta != null && meta.onCellContextMenu != null) {
```
Comment (bakesteve) [inline src/Cell.js]: should check typeof onCellContextMenu is a function rather than not null(ish)

File: src/RowContainer.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import {ContextMenuLayer} from 'react-contextmenu';
+
+class RowContainer extends React.Component {
+  hasContextMenu() {
+    return this.props.contextMenu && React.isValidElement(this.props.contextMenu);
+  }
+
+  renderRowsWithContextMenu() {
+    let newProps = {rowIdx: this.props.rowIdx, idx: this.props.idx};
+    let contextMenu = React.cloneElement(this.props.contextMenu, newProps);
```
Comment (bakesteve) [inline src/RowContainer.js]: not sure what recommended practice is for react-context-menu, but here we will be doing the clone on every row-render, which seems expensive?

**Comment by bakesteve** [comment]
LGTM aside from question on performance impact within row render

**Comment by willgates** [comment]
@bakesteve I fixed checking the type of onCellContextMenu. Re the RowContainer - I imagine the cloning only happens whenever the Canvas re-renders, not on every row render.

File: examples/scripts/example18-context-menu.js [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+var MyContextMenu = React.createClass({
+  onItemClick: function(e, data) {
+    if (this.props.onItemClick) {
+      this.props.onItemClick(e, data);
+    }
+  },
+  render: function() {
+    return (
+      <ContextMenu>
+        <MenuItem data={{action: 'Delete Row', rowIdx: this.props.rowIdx, idx: this.props.idx}} onClick={this.onItemClick}>Delete Row</MenuItem>
```
Comment (malonecj) [inline examples/scripts/example18-context-menu.js]: use an onDeleteRow instead of passing a string to onItemClick

File: src/Canvas.js [hunk 4]
```diff
@@ -282,9 +284,12 @@ const Canvas = React.createClass({
         style={style}
         onScroll={this.onScroll}
         className={joinClasses('react-grid-Canvas', this.props.className, {opaque: this.props.cellMetaData.selected && this.props.cellMetaData.selected.active})}>
-        <div style={{width: this.props.width, overflow: 'hidden'}}>
-          {rows}
-        </div>
+        <RowContainer
```
Comment (malonecj) [inline src/Canvas.js]: Change to `RowsContainer`

File: src/Cell.js [hunk 5]
```diff
@@ -77,22 +77,29 @@ const Cell = React.createClass({
...
+    let meta = this.props.cellMetaData;
+    if (meta != null && typeof(meta.onCellContextMenu) === 'function') {
+      meta.onCellContextMenu({rowIdx: this.props.rowIdx, idx: this.props.idx});
+    }
+  },
+
   onCellDoubleClick() {
     let meta = this.props.cellMetaData;
-    if (meta != null && meta.onCellDoubleClick != null) {
+    if (meta != null && typeof(meta.onCellDoubleClick) === 'function') {
```
Comment (malonecj) [inline src/Cell.js]: :+1:

File: src/RowContainer.js (hunk 2 above)
Comment (malonecj) [inline src/RowContainer.js]: This component is a simple wrapper for all rows. So we will only clone once when the canvas is rerendered. Change the name here to `RowsContainer`

File: examples/scripts/example18-context-menu.js [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+ReactDOM.render(<Example />, mountNode);
+`;
+
+class ContextMenuExample extends React.Component {
+  render() {
+    return (
+      <div>
+        <h3>Context Menu Example</h3>
+        <p>
+          To use a context menu on the grid, create a <code>ReactDataGrid.Menu.ContextMenu</code> and then set the <code>contextMenu</code> prop of the grid to this context menu.
```
Comment (malonecj) [inline examples/scripts/example18-context-menu.js]: Put a link in here giving credit  to the context menu plugin that we are using

**Comment by malonecj** [comment]
Great work @willgates

**Pull request merged**